components/profile-controller/pkg/apis/kubeflow/v1beta1/profile_types.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2019 The Kubeflow Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Plugin is for customize actions on different platform.
+type Plugin struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              *runtime.RawExtension `json:"spec,omitempty"`
+}
+
+// ProfileSpec defines the desired state of Profile
+type ProfileSpec struct {
+	// The profile owner
+	Owner   rbacv1.Subject `json:"owner,omitempty"`
+	Plugins []Plugin       `json:"plugins,omitempty"`
+	// Resourcequota that will be applied to target namespace
+	ResourceQuotaSpec v1.ResourceQuotaSpec `json:"resourcequotaspec,omitempty"`
+}
+
+type ProfileState string
+
+const (
+	ProfileSucceed ProfileState = "Succeed"
```

Comment:
should this be Successful? Should it agree with ApplicationStatus or DeploymentStatus

Reply:
updated